src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala

```diff
@@ -1030,32 +1030,35 @@ trait SilverTransforms extends SparkSessionWrapper {
       .drop("details")
       .withColumnRenamed("type", "state")
 
-    val refinedClusterEventsDFFiltered = refinedClusterEventsDF
-      .withColumn("last_state", first(col("state")).over(orderingWindow))
-      .withColumn("row", row_number().over(orderingWindow))
-      .filter('last_state =!= "TERMINATING" && 'row === 1)
+    val clusterEventsFinal  = if (jrsilverDF.isEmpty || clusterSpec.asDF.isEmpty) {
+      refinedClusterEventsDF
+    }else{
+      val refinedClusterEventsDFFiltered = refinedClusterEventsDF
+        .withColumn("last_state", first(col("state")).over(orderingWindow))
```

Comment:
I didn't catch this last time, but I think this is unnecessary, you can just get the "state" of whichever row ends up being the first in the window.

Reply:
@gueniai  addressed